Included litmusbook redis application statefulset deployers

vibhor995 opened the pull request:
Signed-off-by: vibhor995 <vibhor.kumar@mayadata.io>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:

- Includes redis statefulset application deployment litmusbook.





@giri @dargasudarshan @ksatchit @nsathyaseelan

nsathyaseelan commented on apps/redis/deployers/redis_statefulset.yml:
```diff
@@ -0,0 +1,99 @@
+  - metadata:
+      name: testclaim
+      annotations:
+        volume.beta.kubernetes.io/storage-class: testclass
```
@vibhor995 change the storageclass annotations into storageClassName

nsathyaseelan commented on apps/redis/deployers/test.yml:
```diff
@@ -0,0 +1,135 @@
+
+        - name: Create test specific namespace.
+          shell: kubectl create ns {{ app_ns }}
+          when: app_ns != 'redis'
```
@vibhor995  check the litmus namespace

dargasudarshan commented:
@vibhor995 Can you paste the litmus job output here and also the LR status?

vibhor995 commented:
**OUTPUT**
```

ansible-playbook 2.7.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
No config file found; using defaults

PLAYBOOK: test.yml *************************************************************
1 plays in ./redis/deployers/test.yml

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
task path: /redis/deployers/test.yml:2
ok: [127.0.0.1]
META: ran handlers

TASK [Record test instance/run ID] *********************************************
task path: /redis/deployers/test.yml:12
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /redis/deployers/test.yml:16
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /redis/deployers/test.yml:23
changed: [127.0.0.1] => {"changed": true, "checksum": "796907062aa2eb9a1c3064366f91fac5c5e4d082", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "7fd659efb6bd9007b45b84269d846317", "mode": "0644", "owner": "root", "size": 417, "src": "/root/.ansible/tmp/ansible-tmp-1539064103.1-252156414437916/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
task path: /redis/deployers/test.yml:34
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.945324", "end": "2018-10-09 05:48:25.804948", "failed_when_result": false, "rc": 0, "start": "2018-10-09 05:48:23.859624", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/redis-deployment created", "stdout_lines": ["litmusresult.litmus.io/redis-deployment created"]}

TASK [Check whether the provider storageclass is applied] **********************
task path: /redis/deployers/test.yml:39
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get sc openebs-standard", "delta": "0:00:00.903948", "end": "2018-10-09 05:48:26.907773", "rc": 0, "start": "2018-10-09 05:48:26.003825", "stderr": "", "stderr_lines": [], "stdout": "NAME               PROVISIONER                    AGE\nopenebs-standard   openebs.io/provisioner-iscsi   29d", "stdout_lines": ["NAME               PROVISIONER                    AGE", "openebs-standard   openebs.io/provisioner-iscsi   29d"]}

TASK [Replace the pvc placeholder with provider] *******************************
task path: /redis/deployers/test.yml:43
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
task path: /redis/deployers/test.yml:49
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application replica values from env] *****************************
task path: /redis/deployers/test.yml:55
ok: [127.0.0.1] => {"ansible_facts": {"app_rkey": "replicas", "app_rvalue": "3"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
task path: /redis/deployers/test.yml:60
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application label values from env] *******************************
task path: /redis/deployers/test.yml:66
ok: [127.0.0.1] => {"ansible_facts": {"app_lkey": "app", "app_lvalue": "redis"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
task path: /redis/deployers/test.yml:71
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Create test specific namespace.] *****************************************
task path: /redis/deployers/test.yml:78
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl create ns redis", "delta": "0:00:00.924400", "end": "2018-10-09 05:48:29.080843", "rc": 0, "start": "2018-10-09 05:48:28.156443", "stderr": "", "stderr_lines": [], "stdout": "namespace/redis created", "stdout_lines": ["namespace/redis created"]}

TASK [Checking the status  of test specific namespace.] ************************
task path: /redis/deployers/test.yml:82
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get ns redis -o jsonpath='{.status.phase}'", "delta": "0:00:00.908242", "end": "2018-10-09 05:48:30.184309", "rc": 0, "start": "2018-10-09 05:48:29.276067", "stderr": "", "stderr_lines": [], "stdout": "Active", "stdout_lines": ["Active"]}

TASK [Deploying Redis statefulset] *********************************************
task path: /redis/deployers/test.yml:89
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f redis_statefulset.yml -n redis", "delta": "0:00:01.180352", "end": "2018-10-09 05:48:31.567412", "rc": 0, "start": "2018-10-09 05:48:30.387060", "stderr": "", "stderr_lines": [], "stdout": "statefulset.apps/rd created\nservice/redis created", "stdout_lines": ["statefulset.apps/rd created", "service/redis created"]}

TASK [Obtain the number of replicas.] ******************************************
task path: /redis/deployers/test.yml:94
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get statefulset -n redis -l \"app=redis\" -o custom-columns=:spec.replicas", "delta": "0:00:00.972669", "end": "2018-10-09 05:48:32.785652", "rc": 0, "start": "2018-10-09 05:48:31.812983", "stderr": "", "stderr_lines": [], "stdout": "\n3", "stdout_lines": ["", "3"]}

TASK [Obtain the ready replica count and compare with the replica count.] ******
task path: /redis/deployers/test.yml:101
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (30 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (29 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (28 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (27 retries left).
changed: [127.0.0.1] => {"attempts": 5, "changed": true, "cmd": "kubectl get statefulset -n redis -l \"app=redis\" -o custom-columns=:..readyReplicas", "delta": "0:00:00.917497", "end": "2018-10-09 05:52:38.828192", "rc": 0, "start": "2018-10-09 05:52:37.910695", "stderr": "", "stderr_lines": [], "stdout": "\n3", "stdout_lines": ["", "3"]}

TASK [Setting pass flag] *******************************************************
task path: /redis/deployers/test.yml:110
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /redis/deployers/test.yml:121
changed: [127.0.0.1] => {"changed": true, "checksum": "78528732b441648287945252e5d8871e6ef48bd4", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "49e9f48ac8666c9be3959325d00dbb52", "mode": "0644", "owner": "root", "size": 415, "src": "/root/.ansible/tmp/ansible-tmp-1539064358.98-138778258834784/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
task path: /redis/deployers/test.yml:132
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.208477", "end": "2018-10-09 05:52:42.259346", "failed_when_result": false, "rc": 0, "start": "2018-10-09 05:52:41.050869", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/redis-deployment configured", "stdout_lines": ["litmusresult.litmus.io/redis-deployment configured"]}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=18   changed=14   unreachable=0    failed=0

vibhor995 commented:
**CR output**
```
Name:         redis-deployment
Namespace:
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"litmus.io/v1alpha1","kind":"LitmusResult","metadata":{"annotations":{},"name":"redis-deployment"},"spec":{"testMetadata":{"app":null,"ch...
API Version:  litmus.io/v1alpha1
Kind:         LitmusResult
Metadata:
  Cluster Name:
  Creation Timestamp:  2018-10-09T05:48:25Z
  Generation:          0
  Resource Version:    6836654
  Self Link:           /apis/litmus.io/v1alpha1/redis-deployment
  UID:                 f0a15df3-cb86-11e8-8143-42010a800193
Spec:
  Test Metadata:
    App:        <nil>
    Chaostype:  <nil>
  Test Status:
    Phase:   completed
    Result:  Pass
Events:      <none>

gprasath commented on apps/redis/deployers/run_litmus_test.yml:
```diff
@@ -0,0 +1,51 @@
+
+          - name: PROVIDER_STORAGE_CLASS
+            # Supported values: openebs-standard, local-storage
+            value: openebs-standard
```
use openebs-standalone for statefulsets.

vibhor995 replied:
ok @gprasath

vibhor995 commented:
**output**
```ansible-playbook 2.7.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
No config file found; using defaults

PLAYBOOK: test.yml *************************************************************
1 plays in ./redis/deployers/test.yml

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
task path: /redis/deployers/test.yml:2
ok: [127.0.0.1]
META: ran handlers

TASK [Record test instance/run ID] *********************************************
task path: /redis/deployers/test.yml:12
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /redis/deployers/test.yml:16
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /redis/deployers/test.yml:23
changed: [127.0.0.1] => {"changed": true, "checksum": "796907062aa2eb9a1c3064366f91fac5c5e4d082", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "7fd659efb6bd9007b45b84269d846317", "mode": "0644", "owner": "root", "size": 417, "src": "/root/.ansible/tmp/ansible-tmp-1539149200.33-15008449697331/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
task path: /redis/deployers/test.yml:34
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.242474", "end": "2018-10-10 05:26:42.326643", "failed_when_result": false, "rc": 0, "start": "2018-10-10 05:26:41.084169", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/redis-deployment created", "stdout_lines": ["litmusresult.litmus.io/redis-deployment created"]}

TASK [Check whether the provider storageclass is applied] **********************
task path: /redis/deployers/test.yml:39
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get sc openebs-standalone", "delta": "0:00:00.877323", "end": "2018-10-10 05:26:43.396642", "rc": 0, "start": "2018-10-10 05:26:42.519319", "stderr": "", "stderr_lines": [], "stdout": "NAME                 PROVISIONER                    AGE\nopenebs-standalone   openebs.io/provisioner-iscsi   30d", "stdout_lines": ["NAME                 PROVISIONER                    AGE", "openebs-standalone   openebs.io/provisioner-iscsi   30d"]}

TASK [Replace the pvc placeholder with provider] *******************************
task path: /redis/deployers/test.yml:43
changed: [127.0.0.1] => {"changed": true, "msg": "2 replacements made"}

TASK [Replace the storageclass placeholder with provider] **********************
task path: /redis/deployers/test.yml:49
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application replica values from env] *****************************
task path: /redis/deployers/test.yml:55
ok: [127.0.0.1] => {"ansible_facts": {"app_rkey": "replicas", "app_rvalue": "3"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
task path: /redis/deployers/test.yml:60
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Get the application label values from env] *******************************
task path: /redis/deployers/test.yml:66
ok: [127.0.0.1] => {"ansible_facts": {"app_lkey": "app", "app_lvalue": "redis"}, "changed": false}

TASK [Replace the application label placeholder] *******************************
task path: /redis/deployers/test.yml:71
changed: [127.0.0.1] => {"changed": true, "msg": "3 replacements made"}

TASK [Create test specific namespace.] *****************************************
task path: /redis/deployers/test.yml:78
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl create ns app-redis-ns", "delta": "0:00:00.882537", "end": "2018-10-10 05:26:45.503957", "rc": 0, "start": "2018-10-10 05:26:44.621420", "stderr": "", "stderr_lines": [], "stdout": "namespace/app-redis-ns created", "stdout_lines": ["namespace/app-redis-ns created"]}

TASK [Checking the status  of test specific namespace.] ************************
task path: /redis/deployers/test.yml:82
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get ns app-redis-ns -o jsonpath='{.status.phase}'", "delta": "0:00:00.927614", "end": "2018-10-10 05:26:46.615283", "rc": 0, "start": "2018-10-10 05:26:45.687669", "stderr": "", "stderr_lines": [], "stdout": "Active", "stdout_lines": ["Active"]}

TASK [Deploying Redis statefulset] *********************************************
task path: /redis/deployers/test.yml:89
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f redis_statefulset.yml -n app-redis-ns", "delta": "0:00:00.966330", "end": "2018-10-10 05:26:47.808250", "rc": 0, "start": "2018-10-10 05:26:46.841920", "stderr": "", "stderr_lines": [], "stdout": "statefulset.apps/rd created\nservice/redis created", "stdout_lines": ["statefulset.apps/rd created", "service/redis created"]}

TASK [Obtain the number of replicas.] ******************************************
task path: /redis/deployers/test.yml:94
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get statefulset -n app-redis-ns -l \"app=redis\" -o custom-columns=:spec.replicas", "delta": "0:00:00.919656", "end": "2018-10-10 05:26:49.028778", "rc": 0, "start": "2018-10-10 05:26:48.109122", "stderr": "", "stderr_lines": [], "stdout": "\n3", "stdout_lines": ["", "3"]}

TASK [Obtain the ready replica count and compare with the replica count.] ******
task path: /redis/deployers/test.yml:101
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (30 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (29 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (28 retries left).
FAILED - RETRYING: Obtain the ready replica count and compare with the replica count. (27 retries left).
changed: [127.0.0.1] => {"attempts": 5, "changed": true, "cmd": "kubectl get statefulset -n app-redis-ns -l \"app=redis\" -o custom-columns=:..readyReplicas", "delta": "0:00:00.842154", "end": "2018-10-10 05:30:54.435334", "rc": 0, "start": "2018-10-10 05:30:53.593180", "stderr": "", "stderr_lines": [], "stdout": "\n3", "stdout_lines": ["", "3"]}

TASK [Setting pass flag] *******************************************************
task path: /redis/deployers/test.yml:110
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /redis/deployers/test.yml:121
changed: [127.0.0.1] => {"changed": true, "checksum": "78528732b441648287945252e5d8871e6ef48bd4", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "49e9f48ac8666c9be3959325d00dbb52", "mode": "0644", "owner": "root", "size": 415, "src": "/root/.ansible/tmp/ansible-tmp-1539149454.57-174726983161546/source", "state": "file", "uid": 0}

TASK [Apply the litmus result CR] **********************************************
task path: /redis/deployers/test.yml:132
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:01.019365", "end": "2018-10-10 05:30:57.547606", "failed_when_result": false, "rc": 0, "start": "2018-10-10 05:30:56.528241", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/redis-deployment configured", "stdout_lines": ["litmusresult.litmus.io/redis-deployment configured"]}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=18   changed=14   unreachable=0    failed=0

vibhor995 commented:
**CR output**
```Name:         redis-deployment
Namespace:    
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"litmus.io/v1alpha1","kind":"LitmusResult","metadata":{"annotations":{},"name":"redis-deployment"},"spec":{"testMetadata":{"app":null,"ch...
API Version:  litmus.io/v1alpha1
Kind:         LitmusResult
Metadata:
  Cluster Name:        
  Creation Timestamp:  2018-10-10T05:26:42Z
  Generation:          0
  Resource Version:    6965471
  Self Link:           /apis/litmus.io/v1alpha1/redis-deployment
  UID:                 121c92c0-cc4d-11e8-8143-42010a800193
Spec:
  Test Metadata:
    App:        <nil>
    Chaostype:  <nil>
  Test Status:
    Phase:   completed
    Result:  Pass
Events:      <none>

gprasath commented on apps/redis/deployers/test.yml:
```diff
@@ -0,0 +1,135 @@
+            dest: litmus-result.yaml
+          vars:
+            test: "{{ test_name }}"
+            app: ""
```
Lets pass application name.
